BUG: feature compilation error

## Summary

Error message:
```sh
error[E0004]: non-exhaustive patterns: `&MoonlinkSecretType::S3` not covered
   --> src/moonlink_metadata_store/src/config_utils.rs:143:15
    |
143 |         match &secret_entry.secret_type {
    |               ^^^^^^^^^^^^^^^^^^^^^^^^^ pattern `&MoonlinkSecretType::S3` not covered
    |
note: `MoonlinkSecretType` defined here
   --> /workspaces/moonlink/src/moonlink/src/storage/mooncake_table/table_secret.rs:5:1
    |
5   | pub enum SecretType {
    | ^^^^^^^^^^^^^^^^^^^
...
11  |     S3,
    |     -- not covered
    = note: the matched value is of type `&MoonlinkSecretType`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
    |
178 ~             },
179 +             &MoonlinkSecretType::S3 => todo!()
    |
```
command: 
```sh
cargo test read_one_record_batch -p moonlink_service --features=storage-s3 -- --nocapture
```

--- a/src/moonlink_metadata_store/src/config_utils.rs
+++ b/src/moonlink_metadata_store/src/config_utils.rs
@@ -195,6 +195,7 @@ pub(crate) fn parse_moonlink_table_config(
 /// Recover filesystem config from persisted config and secret.
 ///
 /// For local filesystem, atomic write option is by default disabled, and it's caller's responsibility to enable if necessary.
+#[allow(unreachable_patterns)]
 fn reconstruct_storage_config_from_root(
     root_uri: &str,
     secret_entry: Option<MoonlinkTableSecret>,
@@ -230,6 +231,15 @@ fn reconstruct_storage_config_from_root(
                     root_directory: root_uri.to_string(),
                     atomic_write_dir: None,
                 };
+            }
+
+            _ => {
+                panic!(
+                    "Storage backend {:?} is not supported or feature not enabled. \
+                     Please enable the corresponding feature flag or check your
+            configuration.",
+                    secret_entry.secret_type
+                );
             }
         }
     }

